src/utils/record-completeness/ComputeRecordCompleteness.ts

```diff
@@ -1,92 +1,113 @@
-  nonCoreMandatoryValidationResult = validateRecordCompleteness(nonCoreMandatorySchema, techRecord);
-  return {coreMandatoryValidationResult, nonCoreMandatoryValidationResult};
 };
 
-export const computeRecordCompleteness = (techRecordWrapper: Vehicle, trailerId?: string): string => {
-  let recordCompleteness = RECORD_COMPLETENESS_ENUM.COMPLETE;
-  let isCoreMandatoryValid = true;
-  let isNonCoreMandatoryValid = true;
-  if (!techRecordWrapper.systemNumber) {
-    throw new HTTPError(400, ERRORS.SYSTEM_NUMBER_GENERATION_FAILED);
+function validateMandatoryTechRecordAttributes(vehicle: Vehicle) {
+  const vehicleType = vehicle.techRecord[0].vehicleType as VEHICLE_TYPE;
+  
+  const coreMandatorySchema: ObjectSchema | undefined = coreMandatorySchemaMap.get(vehicleType);
+
+  if (!coreMandatorySchema) {
+    throw new HTTPError(400, ERRORS.VEHICLE_TYPE_ERROR);
   }
-  const generalAttributes = {
-    systemNumber: techRecordWrapper.systemNumber,
-    vin: techRecordWrapper.vin,
-    primaryVrm: techRecordWrapper.primaryVrm,
-    // FIXME: handle trailerId in a better way
-    trailerId
+
+  const nonCoreMandatorySchema: ObjectSchema | undefined = nonCoreMandatorySchemaMap.get(vehicleType);
+
+  const techRecord = vehicleType === VEHICLE_TYPE.TRL
+    ? { ...vehicle.techRecord[0], primaryVrm: vehicle.primaryVrm }
+    : vehicle.techRecord[0];
+
+  return {
+    coreValidationResult: coreMandatorySchema.validate(techRecord, {stripUnknown: true}),
+    nonCoreValidationResult: nonCoreMandatorySchema?.validate(techRecord, {stripUnknown: true})
   };
-  const vehicleType = techRecordWrapper.techRecord[0].vehicleType;
-  if (!vehicleType) {
-    return RECORD_COMPLETENESS_ENUM.SKELETON;
-  }
-  const generalErrors = validateVehicleAttributes(vehicleType, generalAttributes)?.error;
+};
 
-  if (generalErrors) {
-    console.log("general errors: ",generalErrors);
-    return RECORD_COMPLETENESS_ENUM.SKELETON;
-  }
+const coreMandatorySchemaMap = new Map<VEHICLE_TYPE, any>([
+  [VEHICLE_TYPE.PSV,        psvCoreMandatorySchema],
+  [VEHICLE_TYPE.HGV,        hgvCoreMandatorySchema],
+  [VEHICLE_TYPE.TRL,        trlCoreMandatorySchema],
+  [VEHICLE_TYPE.LGV,        lgvCoreMandatorySchema],
+  [VEHICLE_TYPE.CAR,        carCoreMandatorySchema],
+  [VEHICLE_TYPE.MOTORCYCLE, motorcycleCoreMandatorySchema]
+]);
 
-  const mandatoryAttributesValidationResult = validateCoreAndNonCoreMandatoryTechRecordAttributes(vehicleType, techRecordWrapper);
-  const mandatoryErrors = mandatoryAttributesValidationResult.coreMandatoryValidationResult?.error;
-  isCoreMandatoryValid = !mandatoryErrors;
-  isNonCoreMandatoryValid = !mandatoryAttributesValidationResult.nonCoreMandatoryValidationResult?.error;
+const nonCoreMandatorySchemaMap = new Map<VEHICLE_TYPE, any>([
+  [VEHICLE_TYPE.PSV, psvNonCoreMandatorySchema],
+  [VEHICLE_TYPE.HGV, hgvNonCoreMandatorySchema],
+  [VEHICLE_TYPE.TRL, trlNonCoreMandatorySchema]
+]);
```

Comment:
I am a big fan of the `Record` utility type, which could achieve a similar result along with the `Partial` utility type for the `nonCoreMandatorySchemaMap`. Trying to find some sources about the speed of Map vs objects (which `Record` compiles to). [this source](https://www.zhenghao.io/posts/object-vs-map) and [this one](https://stackoverflow.com/questions/18541940/map-vs-object-in-javascript) points to map being better for all CRUD operations, where as objects being better for read operations. Don't mind it staying a map though

Reply:
I would upvote maps for most things, being armotised constant in time complexity for the majority 😄

Reply:
This is just a few lines of code that deal with a list of things that are less than 10. Speed is really not an issue here.

`Map`, `Record` or `Partial` would all be blisteringly fast. I will leave these as is for now as they are very easy to understand, but thanks for the suggestion. I know of `Partial` but I am yet to use `Record` in my code.